### PR TITLE
Enforce namespace in FilesystemClient

### PR DESCRIPTION
**Pull request by robin-nitrokey**
This patch adds a check to FilesystemClient that makes sure that clients cannot read out files that belong to other clients or that are not in the data directory for the client.

This is a breaking change as the Error enum is extended.

Fixes https://github.com/trussed-dev/trussed/issues/65

### CODE REVIEW

File: src/mechanisms/hmacsha256.rs [hunk 1]
```diff
@@ -18,7 +18,10 @@ impl DeriveKey for super::HmacSha256 {
         if !matches!(key.kind, key::Kind::Symmetric(..) | key::Kind::Shared(..)) {
             // We have to disable this check for compatibility with fido-authenticator, see:
             // https://github.com/solokeys/fido-authenticator/issues/21
-            warn!("derive_key for hmacsha256 called with invalid key kind ({:?})", key.kind);
+            warn!(
+                "derive_key for hmacsha256 called with invalid key kind ({:?})",
+                key.kind
+            );
```
Comment (szszszsz) [inline src/mechanisms/hmacsha256.rs:24]: This change looks unrelated

Comment (robin-nitrokey) [reply src/mechanisms/hmacsha256.rs:24]: moved to: https://github.com/trussed-dev/trussed/pull/69

Comment (szszszsz) [reply src/mechanisms/hmacsha256.rs:24]: It's still here, unless there is something with my GH view.

Comment (robin-nitrokey) [reply src/mechanisms/hmacsha256.rs:24]: Yes, this PR has not been rebased yet.  But if you compare to current main, there are no changes to `src/mechanisms/hmacsha256.rs` anymore.

Comment (szszszsz) [reply src/mechanisms/hmacsha256.rs:24]: Right. Resolving.

**Comment by robin-nitrokey** [comment]
Rebased onto 88de4c6b3ac8c94534aad98efbed02af8eaec70c.

**Comment by robin-nitrokey** [comment]
Rebased and changed to just forbid `..` for performance reasons.
